### PR TITLE
Fixes Gradle build in Scala projects

### LINKED ISSUE #6316: A generated Scala project with Gradle doesn't build
**Describe the bug**
When generating a project like so:

```bash
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create     \
-DprojectGroupId=org.acme     \
-DprojectArtifactId=gradle-scala-test \     
-DclassName="org.acme.GreetingResource"\    
 -Dpath="/greeting"\
-Dextensions="scala"\ 
-DplatformArtifactId=quarkus-bom \
-DbuildTool=gradle
```

and then trying to build it like so:

```
gradle build -x test
```

I get:

```
> Task :compileScala FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileScala'.
> Cannot infer Scala class path because no Scala library Jar was found. Does root project 'gradle-scala-test' declare dependency to scala-library? Searched classpath: file collection.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
1 actionable task: 1 executed
```

In Windows, a different error is shown in the CI server: https://scans.gradle.com/s/turf2ey7xcl2o/tests/dqwcqqtprpdqc-tvfi6t4p2x5we

**Environment (please complete the following information):**
 - Quarkus version or git rev: latest `master`

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #6316

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java [hunk 1]
```diff
@@ -100,4 +100,16 @@ public AppArtifact getAppArtifact() {
     public AppModelResolver resolveAppModel() {
         return new AppModelGradleResolver(project);
     }
+
+    /**
+     * Returns the last file from the specified {@link FileCollection}.
+     * Needed for the Scala plugin.
```
Comment (geoand) [inline devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java]: Now that is weird :)

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java [hunk 2]
```diff
@@ -74,13 +74,16 @@ public void canBuild(SourceType sourceType) throws IOException {
     }
 
     private void createProject(SourceType sourceType) throws IOException {
+        Map<String,Object> context = new HashMap<>();
+        context.put("path", "/greeting");
```
Comment (geoand) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java]: I don't really understand what this is necessary. Can you please elaborate?

**Comment by gastaldi** [comment]
This is how I tell the ProjectGenerator that I need a JAX-RS endpoint too.

Em Seg, 23 de dez de 2019 03:15, Georgios Andrianakis <
notifications@github.com> escreveu:

**Comment by geoand** [comment]
Oh, ok makes sense

**Review by geoand — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Pull request merged**